src/conv/problem_description.cpp

```diff
@@ -59,6 +60,41 @@ std::ostream& operator<<(std::ostream& stream, std::function<void(std::ostream&)
     return stream;
 }
 
+void ProblemDescription::HeuristicUpdateLayouts()
+{
+    const std::string labels = tensor_layout_get_default(in_layout.size());
+
+    std::set<std::string> input_layouts;
+    std::set<std::string> output_layouts;
+    std::set<std::string> weight_layouts;
+
+    static const std::vector<std::string> supported_layouts = {"NCHW", "NHWC", "NCDHW"};
+    for(const std::string& layout : supported_layouts)
+    {
+        // Skip layouts that doesn't match dimension sizes
+        if(layout.size() != labels.size())
+            continue;
+
+        if(in.IsLayout(labels, layout))
+            input_layouts.insert(layout);
+        if(out.IsLayout(labels, layout))
+            output_layouts.insert(layout);
+        if(weights.IsLayout(labels, layout))
+            weight_layouts.insert(layout);
+
+        // If we have found consistent layout among 3 tensors, assign them
+        if((input_layouts.count(layout) == 1) && (weight_layouts.count(layout) == 1) &&
+           (output_layouts.count(layout) == 1))
```

Comment:
```suggestion
        if(in.IsLayout(labels, layout) && out.IsLayout(labels, layout) && weights.IsLayout(labels, layout))
```
and remove 
```
    std::set<std::string> input_layouts;
    std::set<std::string> output_layouts;
    std::set<std::string> weight_layouts;
```

Reply:
@jerryyin what do you think about this change?

Reply:
@carlushuang Thanks for ping me. Yep, I think @atamazov 's suggestion is good. I don't think storing all possible layout in the set is necessary. It sets the first consistent layout according to `supported_layouts` and be done.

Reply:
OK, I'll change based on this review